cmd/cloud/server.go

```diff
@@ -50,20 +53,43 @@ var serverCmd = &cobra.Command{
 		s3StateStore, _ := command.Flags().GetString("state-store")
 		logger.Infof("Using state store %s", s3StateStore)
 
+		// Limit the number of concurrent background jobs within this instance.
+		jobs, _ := command.Flags().GetInt("jobs")
+		workers := semaphore.NewWeighted(int64(jobs))
+		defer func() {
+			if err := workers.Acquire(context.TODO(), int64(jobs)); err != nil {
+				logger.WithError(err).Error("failed to shut down worker pool")
+			}
+		}()
+
+		// Setup the provisioner for actually effecting changes to clusters.
+		kopsProvisioner := provisioner.NewKopsProvisioner(
+			clusterRootDir,
+			s3StateStore,
+			logger,
+		)
+
+		// Setup the supervisor to effect any requested changes. It is wrapped in a
+		// scheduler to trigger it periodically in addition to being poked by the API
+		// layer.
+		poll, _ := command.Flags().GetInt("poll")
+		if poll == 0 {
+			logger.Info("Switching scheduler to manual mode only")
```

Comment:
Hmm, this behavior is bit strange to me. With `poll` set to 0, the provisioning server will act on requests only if it has a non-busy job at the time the request is made. If the provisioning server is completely busy when another API request is made then the cluster objects will have their state updated, but no actions will be taken on them. Obviously, if another provisioning server is watching the cluster store objects that isn't set to `poll=0` then it will act on them, but I am wondering if we want any provisioning servers to behave like this when `poll=0`?

I could see this being useful if manual mode returned an error if it was completely busy and didn't update an cluster states at all so that the caller knew that another attempt was needed later.

Thoughts?

Reply:
You're right: the dissonance here is wrong. My primary use case was to force the server into not running "background" jobs, which can be really handy when pointing at a shared database for debugging but not wanting to start working on outstanding jobs.

However, the behaviour I have implemented that supports a "manual" poke makes that weird. Maybe if it had poked only the item in question, but I think I'll just rip the manual mode out and disable the scheduler altogether if `poll` is `0`.